server/chat-plugins/helptickets.ts

```diff
@@ -1210,6 +1264,7 @@ export const commands: ChatCommands = {
 		close(target, room, user) {
 			if (!target) return this.parse(`/help helpticket close`);
 			let result = !(this.splitTarget(target) === 'false');
+			const tickets = getTickets();
 			const ticket = tickets[toID(this.inputUsername)];
```

Comment:
For instance, in `ticket close`, too. You set `ticket.open = false` here, which does nothing.

Reply:
This code also contains the assumption that the corresponding Help room is ``Rooms.get(`help-${ticket.userid}`)`` which is also wrong.

Actually, there are a lot of false assumptions here which this refactor is revealing. Should tickets be retained beyond the Help room being deleted? I might need to investigate personally.

Reply:
I don't think it should be saved beyond the ticket deletion. 
Why would the help room not be help-ticket.userid, though?

Reply:
I pushed the readonly changes and tried to change the assumptions i noticed, is that better-ish?

Reply:
`/ht list` contains a lot of old tickets where the room no longer exists. Would those just all be gone in your refactor?

Reply:
They would, yes. Is that not desirable? I thought that's what you were going for - I can remove it if it's not.

Reply:
When I suggested that ticket state could be autogenerated from existing tickets, I didn't realize we kept old ticket state past when we kept old tickets. I wouldn't have suggested that if I knew.

I'm not _opposed_ to changing this, but it definitely wasn't intentional on my part.

Reply:
I asked Staff, they agree it's no big deal to get rid of expired tickets.